### ISSUE TITLE
Test failure: ResponseCompression_AppCompression

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/ServerTests#145]
This test [fails](http://aspnetci/viewLog.html?buildId=557737&buildTypeId=XPlat_Windows_Win8_Universe) occasionally with the following error:
```
Assert.Equal() Failure
          ↓ (pos 0)
Expected: Running
Actual:   <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML ···
          ↑ (pos 0)
   at ServerComparison.FunctionalTests.ResponseCompressionTests.ResponseCompression(TestVariant variant, Func`3 scenario, Boolean hostCompression, String testName) in /_/test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs:line 162
--- End of stack trace from previous location where exception was thrown ---
------- Stdout: -------
| [0.001s] TestLifetime Information: Starting test ResponseCompression_AppCompression-Server: IISExpress, TFM: net461, Type: Standalone, Arch: x64, ANCM: V1, Host: OutOfProcess at 2018-09-20T18:48:19
| [0.001s] TestLifetime Information: Starting test ResponseCompression_AppCompression_IISExpress_net461_x64_Standalone at 2018-09-20T18:48:19
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Deploying [Variation] :: ServerType=IISExpress, Runtime=Clr, Arch=x64, BaseUrlHint=, Publish=False
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Executing: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe 
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: ContentRoot: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites
| [0.002s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Attempting to start IIS Express on port: 54176
| [0.006s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Executing command : C:\Program Files\IIS Express\iisexpress.exe /site:HttpTestSite /config:C:\BuildAgent\temp\buildTmp\tmp89EA.tmp /trace:error /systray:false
| [0.006s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: SET ASPNETCORE_ENVIRONMENT=ResponseCompression
| [0.006s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: SET ASPNETCORE_DETAILEDERRORS=true
| [0.006s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: SET LAUNCHER_PATH=C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe
| [0.006s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: SET LAUNCHER_ARGS=
| [0.006s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: SET ASPNETCORE_CONTENTROOT=C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites
| [0.008s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress Process 16468 started
| [0.063s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Starting IIS Express ...
| [0.077s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Initializing the W3 Server Started CTC = 214796812 
| [0.082s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: W3 Server initializing WinSock.  CTC = 214796828 
| [0.082s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: W3 Server WinSock initialized.  CTC = 214796828 
| [0.082s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: W3 Server ThreadPool initialized (ipm has signalled).  CTC = 214796828 
| [0.103s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Start listenerChannel http:0
| [0.104s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Successfully registered URL "http://localhost:54176/" for site "HttpTestSite" application "/"
| [0.104s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Registration completed for site "HttpTestSite"
| [0.104s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: AppPool 'IISExpressAppPool' initialized
| [0.104s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: InitComplete event signalled
| [0.104s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: IIS Express is running.
| [0.105s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Started iisexpress successfully. Process Id : 16468, Port: 54176
| [0.105s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: Application ready at URL: http://localhost:54176/
| [0.105s] ResponseCompression Warning: Retry count 1..
| [0.107s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: IncrementMessages called
| [0.108s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Request started: "GET" http://localhost:54176/
| [0.759s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISExpressDeployer Information: iisexpress stdout: Request ended: http://localhost:54176/ with HTTP status 502.3
| [0.761s] ResponseCompression Warning: StatusCode: 502, ReasonPhrase: 'Bad Gateway', Version: 1.1, Content: System.Net.Http.HttpConnection+HttpConnectionResponseContent, Headers:
|                                       {
|                                         Cache-Control: private
|                                         Server: Microsoft-IIS/10.0
|                                         X-SourceFiles: =?UTF-8?B?QzpcQnVpbGRBZ2VudFx3b3JrXDMzYmRmYzFjYWU3YjJhMzhcbW9kdWxlc1xTZXJ2ZXJUZXN0c1x0ZXN0XFNlcnZlckNvbXBhcmlzb24uVGVzdFNpdGVz?=
|         
...
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on master.

CC @Eilon (because the bot doesn't know who else to pick)

This issue was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
HelloWorld [failed with about the same error](http://aspnetci/viewLog.html?buildId=557737&buildTypeId=XPlat_Windows_Win8_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
HelloWorld [failed with about the same error](http://aspnetci/viewLog.html?buildId=558038&buildTypeId=XPlat_Windows_Win8_Universe) on master.

This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by Tratcher** [comment]
Same issue as elsehwere: https://github.com/aspnet/Home/issues/3544

```
System.IO.FileLoadException: Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
| [0.523s] Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer Information: C:\BuildAgent\work\33bdfc1cae7b2a38\modules\ServerTests\test\ServerComparison.TestSites\bin\Release\net461\ServerComparison.TestSites.exe stdout: File name: 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
```

**Comment by aspnet-hello** [comment]
There were 12 failures [with about the same error](http://aspnetci/viewLog.html?buildId=558745&buildTypeId=Lite_UniverseTest) on master:
- HelloWorld
- ResponseCompression_AppCompression
- ResponseCompression_NoCompression
- ResponseCompression_HostCompression
- HelloWorld
- HelloWorld
- ResponseCompression_HostCompression
- ResponseCompression_AppCompression
- NtlmAuthentication
- HelloWorld
- (And 2 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 12 failures [with about the same error](http://aspnetci/viewLog.html?buildId=559368&buildTypeId=XPlat_Windows_Win8_Universe) on master:
- HelloWorld
- ResponseCompression_HostCompression
- NtlmAuthentication
- ResponseCompression_HostCompression
- HelloWorld
- ResponseCompression_NoCompression
- ResponseCompression_NoCompression
- ResponseCompression_AppCompression
- NtlmAuthentication
- HelloWorld
- (And 2 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 12 failures [with about the same error](http://aspnetci/viewLog.html?buildId=560048&buildTypeId=XPlat_Windows_Win8_Universe) on master:
- ResponseCompression_NoCompression
- ResponseCompression_NoCompression
- NtlmAuthentication
- HelloWorld
- HelloWorld
- HelloWorld
- ResponseCompression_AppCompression
- ResponseCompression_HostCompression
- ResponseCompression_AppCompression
- NtlmAuthentication
- (And 2 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 12 failures [with about the same error](http://aspnetci/viewLog.html?buildId=560221&buildTypeId=XPlat_Windows_Win7_Universe) on master:
- HelloWorld
- HelloWorld
- HelloWorld
- ResponseCompression_AppCompression
- ResponseCompression_NoCompression
- HelloWorld
- NtlmAuthentication
- ResponseCompression_HostCompression
- ResponseCompression_HostCompression
- ResponseCompression_NoCompression
- (And 2 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by aspnet-hello** [comment]
There were 12 failures [with about the same error](http://aspnetci/viewLog.html?buildId=561382&buildTypeId=Lite_UniverseTest) on master:
- HelloWorld
- ResponseCompression_NoCompression
- HelloWorld
- HelloWorld
- NtlmAuthentication
- ResponseCompression_HostCompression
- ResponseCompression_NoCompression
- ResponseCompression_AppCompression
- ResponseCompression_AppCompression
- HelloWorld
- (And 2 more test failures in this build)


This comment was made automatically. If there is a problem contact ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
This was solved when @natemcmaster fixed aspnet/Home#3544.
